middleware/grpc_cluster_test.go

```diff
@@ -0,0 +1,106 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/go-kit/log"
+	"github.com/gogo/status"
+
+	"github.com/grafana/dskit/httpgrpc"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+)
+
+func TestClusterUnaryClientInterceptor(t *testing.T) {
+	testCases := map[string]struct {
+		cluster                    string
+		expectedClusterFromContext string
+	}{
+		"no cluster info sets no cluster info in context": {
+			cluster:                    "",
+			expectedClusterFromContext: "",
+		},
+		"if cluster info is set, it should be propagated to invoker": {
+			cluster:                    "cluster",
+			expectedClusterFromContext: "cluster",
+		},
+	}
+	verify := func(ctx context.Context, expectedCluster string) {
+		md, ok := metadata.FromOutgoingContext(ctx)
+		require.True(t, ok)
+		clusterIDs, ok := md[MetadataClusterKey]
+		require.True(t, ok)
+		require.Len(t, clusterIDs, 1)
+		require.Equal(t, expectedCluster, clusterIDs[0])
+	}
+	for testName, testCase := range testCases {
+		t.Run(testName, func(t *testing.T) {
+			interceptor := ClusterUnaryClientInterceptor(testCase.cluster)
+			invoker := func(ctx context.Context, _ string, _, _ any, _ *grpc.ClientConn, _ ...grpc.CallOption) error {
+				if testCase.expectedClusterFromContext != "" {
+					verify(ctx, testCase.expectedClusterFromContext)
+				}
```

Comment:
Shouldn't we also verify the inverse, i.e. that no cluster is defined in the metadata when none is expected?